packages/sandpack-core/src/resolver/resolver.test.ts

```diff
@@ -503,6 +503,30 @@ describe('resolve', () => {
         '/node_modules/package-exports/src/utils/test/index.js'
       );
     });
+
+    it('should ignore the closest package.json and aliases in it when exports are available in the root', () => {
+      const resolved = resolveSync('@emotion/react/jsx-runtime', {
+        filename: '/foo.js',
+        extensions: ['.ts', '.tsx', '.js', '.jsx'],
+        isFile,
+        readFile,
+      });
+      expect(resolved).toBe(
+        '/node_modules/@emotion/react/jsx-runtime/dist/emotion-react-jsx-runtime.cjs.js'
+      );
+    });
+
+    it('should always use exports from the root of the package and not from the closest package.json', () => {
```

Comment:
Is there a specification you can point to for this test? I always thought having a package.json means that is the root of said package. So it should always use the closest package.json no? And how can we reliably detect the root package.json, in case of pnpm or something this logic no longer means it will be under `node_modules/pkg/package.json` but instead somewhere `.store/pkg@version/node_modules/pkg/package.json`

Reply:
For other reviewers, you can have package layouts like this:
```ts
pkg-a/package.json
pkg-a/extra/package.json
```

When requesting `pkg-a/extra` we have to:
- first check if there is `pkg-a/package.json#exports`
- if it's there then it becomes the only source of information about available module specifiers within the package etc
- it that's not there then `pkg-a/extra/package.json` should be used as the possible source of information, main fields like `pkg-a/extra/package.json#main`, `pkg-a/extra/package.json#module`, `pkg-a/extra/package.json#browser` might have to be interpreted and tried

At the same time, `pkg-a/extra/package.json#exports` should **always** be ignored. `exports` is reserved for the "root" `package.json`